src/orion/algo/space/configspace.py

```diff
@@ -0,0 +1,261 @@
+                default_value=dim.default_value,
+                q=_qantization(dim),
+                log=False,
+                lower=dim.low if hasattr(dim, "low") else None,
+                upper=dim.high if hasattr(dim, "high") else None,
+            )
+
+            return NormalIntegerHyperparameter(**kwargs)
+
+        raise UnsupportedPrior(f'Prior "{dim.prior_name}" is not supported')
+
+    def categorical(self, dim: Categorical) -> Optional[CategoricalHyperparameter]:
+        """Convert a categorical dimension into a configspace equivalent"""
+        return CategoricalHyperparameter(
+            name=dim.name,
+            choices=dim.categories,
+            weights=dim._probs,
+        )
+
+    def fidelity(self, dim: Fidelity) -> None:
+        """Ignores fidelity dimension as configspace does not have an equivalent"""
+        return None
+
+    def space(self, space: Space) -> ConfigurationSpace:
+        """Convert orion space to configspace"""
+        cspace = ConfigurationSpace()
+        dims = []
+
+        for _, dim in space.items():
+            cdim = self.visit(dim)
+
+            if cdim:
+                dims.append(cdim)
+
+        cspace.add_hyperparameters(dims)
+        return cspace
+
+
+def to_configspace(space: Space) -> ConfigurationSpace:
+    """Convert orion space to configspace
+
+    Notes
+    -----
+    ``ConfigurationSpace`` will set its own default values
+    if not set inside ``Space``
+
+    """
+    conversion = ToConfigSpace()
+    return conversion.space(space)
+
+
+@singledispatch
+def to_oriondim(dim: Hyperparameter) -> Dimension:
+    """Convert a config space hyperparameter to an orion dimension"""
+    raise NotImplementedError(f"Dimension {dim} is not supported by Orion")
+
+
+@to_oriondim.register
+def from_categorical(dim: CategoricalHyperparameter) -> Dimension:
+    """Builds a categorical dimension from a categorical hyperparameter"""
+    choices = {k: w for k, w in zip(dim.choices, dim.probabilities)}
+    return Categorical(dim.name, choices)
+
+
+@to_oriondim.register(UniformIntegerHyperparameter)
+@to_oriondim.register(UniformFloatHyperparameter)
+def from_uniform(dim: Hyperparameter) -> Dimension:
+    """Builds a uniform dimension from a uniform hyperparameter"""
+
+    klass = Integer
+    args = []
+    kwargs = dict(
+        # NOTE: Config space always has a config value
+        # so orion-space would get it as well
+        default_value=dim.default_value
+    )
+
+    if isinstance(dim, UniformFloatHyperparameter):
+        klass = Real
+    else:
+        kwargs["precision"] = int(-log10(dim.q)) if dim.q else 4
+
+    dist = "uniform"
+    args.append(dim.lower)
+    args.append(dim.upper)
+
+    if dim.log:
+        dist = "reciprocal"
+
+    return klass(dim.name, dist, *args, **kwargs)
+
+
+@to_oriondim.register(NormalFloatHyperparameter)
+@to_oriondim.register(NormalIntegerHyperparameter)
+def from_normal(dim: Hyperparameter) -> Dimension:
```

Comment:
Nit: Narrowing the type signature:
```suggestion
def from_normal(dim: Hyperparameter) -> Integer | Real:
```